experiment/src/org/labkey/experiment/api/ExpDataClassDataTableImpl.java

```
@@ -1370,6 +1370,32 @@ else if (name != null)
             return dataRow;
         }
 
+
+        @Override
+        protected Map<String, Object> updateRow(User user, Container container, Map<String, Object> row, @NotNull Map<String, Object> oldRow, boolean allowOwner, boolean retainCreation)
+                throws InvalidKeyException, ValidationException, QueryUpdateServiceException, SQLException
+        {
+            Map<String, Object> result = super.updateRow(user, container, row, oldRow, allowOwner, retainCreation);
+
+            // add MaterialInput/DataInputs field from parent alias
```

I'm unclear why this wasn't necessary before. Can you enlighten me?

---

There was a bug that row by row update using alias gets silently ignored. It's reproducible only from API, since our UI will never send the request using alias, and file import will never hit the row by row code. Due to its narrow repro, I didn't create a separate issue for it.